lib/src/internal/http/http_route.dart

```diff
@@ -0,0 +1,116 @@
+
+  void webhooks({String? id, String? token}) => _httpRouteParts.add(HttpRoutePart("webhooks", [
+        if (id != null) HttpRouteParam(id, isMajor: token != null),
+        if (token != null) HttpRouteParam(token, isMajor: id != null),
+      ]));
+
+  void reactions({String? emoji, String? userId}) => add(HttpRoutePart("reactions", [
+        if (emoji != null) HttpRouteParam(emoji),
+        if (userId != null) HttpRouteParam(userId),
+      ]));
+
+  void emojis({String? id}) => add(HttpRoutePart("emojis", [if (id != null) HttpRouteParam(id)]));
+
+  void roles({String? id}) => add(HttpRoutePart("roles", [if (id != null) HttpRouteParam(id)]));
+
+  void members({String? id}) => add(HttpRoutePart("members", [if (id != null) HttpRouteParam(id)]));
+
+  void bans({String? id}) => add(HttpRoutePart("bans", [if (id != null) HttpRouteParam(id)]));
+
+  void users({String? id}) => add(HttpRoutePart("users", [if (id != null) HttpRouteParam(id)]));
+
+  void permissions({String? id}) => add(HttpRoutePart("permissions", [if (id != null) HttpRouteParam(id)]));
+
+  void messages({String? id}) => add(HttpRoutePart("messages", [if (id != null) HttpRouteParam(id)]));
+
+  void pins({String? id}) => add(HttpRoutePart("pins", [if (id != null) HttpRouteParam(id)]));
+
+  void invites({String? id}) => add(HttpRoutePart("invites", [if (id != null) HttpRouteParam(id)]));
+
+  void applications({String? id}) => add(HttpRoutePart("applications", [if (id != null) HttpRouteParam(id)]));
+
+  void stageinstances({String? id}) => add(HttpRoutePart("stage-instances", [if (id != null) HttpRouteParam(id)]));
+
+  void threadMembers({String? id}) => add(HttpRoutePart("thread-members", [if (id != null) HttpRouteParam(id)]));
+
+  void stickers({String? id}) => add(HttpRoutePart("stickers", [if (id != null) HttpRouteParam(id)]));
+
+  void avatars() => add(HttpRoutePart("avatars"));
+
+  void scheduledEvents({String? id}) => add(HttpRoutePart("scheduled-events", [if (id != null) HttpRouteParam(id)]));
+
+  void prune() => add(HttpRoutePart("prune"));
+
+  void nick() => add(HttpRoutePart("nick"));
+
+  void auditlogs() => add(HttpRoutePart("audit-logs"));
+
+  void regions() => add(HttpRoutePart("regions"));
+
+  void search() => add(HttpRoutePart("search"));
+
+  void bulkdelete() => add(HttpRoutePart("bulk-delete"));
+
+  void typing() => add(HttpRoutePart("typing"));
+
+  void crosspost() => add(HttpRoutePart("crosspost"));
+
+  void threads() => add(HttpRoutePart("threads"));
+
+  void gateway() => add(HttpRoutePart("gateway"));
+
+  void bot() => add(HttpRoutePart("bot"));
+
+  void oauth2() => add(HttpRoutePart("oauth2"));
+
+  void preview() => add(HttpRoutePart("preview"));
+
+  void active() => add(HttpRoutePart("active"));
+
+  void archived() => add(HttpRoutePart("archived"));
+
+  void private() => add(HttpRoutePart("private"));
+
+  void public() => add(HttpRoutePart("public"));
+
+  void stickerpacks() => add(HttpRoutePart("sticker-packs"));
```

Comment:
Can all of these methods be documented?

Reply:
Do you have an example of documentation? The individual builder methods seem self explanatory to me. I have already added a class level doc at the top. 

It is currently true that any path parameter that needs to be marked `isMajor` is always marked `isMajor`, although I didn't want to hard code that as an assumption. These helper builder methods include that assumption and make `HttpRoute` declarations less verbose for the 100+ uses of this class.

Reply:
The documentation for each would probably be very basic, and it more so we can have a more complete documentation coverage of the package.

An example for the `stickerpacks()` method would be something like this:
```
Adds the [`sticker-packs`](https://discord.com/developers/docs/resources/sticker#list-nitro-sticker-packs) part to this [HttpRoute].
```

Reply:
Now that Im thinking about it. It isnt that much of a problem. There are few methods like `public` which could be confusing but sticker guilds and similar would be good to leave without docs